### DISCUSSION TITLE
Yoruba Stopwords

### DISCUSSION THREAD

**Discussion by chrisemezue** [masakhane-io/masakhanePreprocessor#12] [african-stopwords]
Discussion on functional words in Yoruba and stopwords

**Comment by Olanrewajusamuel** [comment]
Most of the words stated are functional words. Yorùbá is less inflectional which means that each word actually carry information to make a complete sense. 

Also, many times, ní could seem to be omitted but it only changed to /l/ as shown in examples (1):
Examples (1)
a. Olú ni owo = Olú lowo (Olu has money)
b. Adé gbé e ni alẹ = Adé gbé lalẹ (Ade carried it in the night)
c. Eji le ni ogún = Ejilelogun (Two more than twenty)

The idea is that when the vowel after /n/ is a nasal, it will retain /n/ but when the vowel is deleted and not nasal, it becomes /l/ (Umfomata, 1988).

But we could have the /n/ retained when it shows the location of something (preposition) as in:
d. Adé gbé ni Ile = Adé gbé Nile (Ade carried it at home)

Further, ó which many believed to be "third person pronoun" in Yoruba has been argued to be high tone syllable (Awobuluyi, 1967) or tone morph.
Ó might be a stopwords not on the ground of grammar but that it doesn't carry any weight than being the carrier of the high tone.

**Reply by chrisemezue** [reply]
> Most of the words stated are functional words. 

These are the words he's referring to: https://github.com/masakhane-io/masakhane-preprocessing/blob/main/african-stopwords/languages/yo.txt

**Reply by shmuhammadd** [reply]
Hi @chrisemezue we also have some stop-word list for Hausa, Igbo, Yoruba and Pidgin from [NaijaSenti](https://github.com/hausanlp/NaijaSenti) project.

**Reply by chrisemezue** [reply]
That's amazing @shmuhammad2004 
I see them [here](https://github.com/hausanlp/NaijaSenti/tree/main/data/stopwords).
Are they open-source? can we use them?

**Reply by shmuhammadd** [reply]
@chrisemezue yes, sure. They are all open source.
